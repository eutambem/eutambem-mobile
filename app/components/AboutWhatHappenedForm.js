import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import appStyles from '../styles';

import { Button, Picker, DatePicker, TextInput, CheckBoxGroup, PlaceInput } from './common';

import { MAX_TEXT_INPUT_LENGTH } from '../constants';

import { formWrapper, loading } from './hoc';

export class AboutWhatHappenedForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      date: null,
      description: '',
      advice: '',
      establishment: {},
    };
  }

  onEstablishmentChange = (item) => {
    this.setState({ establishment: { value: item.id, label: item.label } });
  }

  formData() {
    return {
      description: this.state.description,
    }
  }

  render() {
    return (
      <View>
        <PlaceInput
          placeholder="Empresa em que ocorreu"
          value={this.state.establishment.label}
          onValueChange={this.onEstablishmentChange}
          types={['establishment']}
          navigation={this.props.navigation}
          required
          placeScreenTitle="Informe a Empresa"
        />
        <Picker
          required
          placeholder="Tipo de assédio"
          items={this.props.data.formOptions.harassment_type_options}
        />
        <TextInput
          multiline
          required
          placeholder="Conte-nos mais sobre o ocorrido"
          onChangeText={description => this.setState({ description })}
          value={this.state.description}
        />
        <DatePicker
          date={this.state.date}
          required
          placeholder="Quando ocorreu"
          maxDate={new Date(Date.now())}
          onDateChange={date => this.setState({ date })}
        />
        <CheckBoxGroup
          label="Você tomou alguma providência com relação ao ocorrido?"
          options={this.props.data.formOptions.followup_actions_options}
        />
        <Picker
          required
          placeholder="Você recomendaria essa empresa depois do ocorrido?"
          items={this.props.data.formOptions.yes_no_optional_options}
        />
        <TextInput
          style={[appStyles.input, appStyles.multilineInput]}
          multiline
          autoGrow
          maxLength={MAX_TEXT_INPUT_LENGTH}
          placeholder="Qual seu conselho para os gestores?"
          onChangeText={advice => this.setState({ advice })}
          value={this.state.advice}
        />
        <Button
          onPress={() => this.props.navigation.navigate('AboutYou', { formData: this.formData() })}
          title="Prosseguir"
        />
      </View>
    );
  }
}

const valueLabelShape = PropTypes.arrayOf(PropTypes.shape({
  value: PropTypes.string,
  label: PropTypes.string,
})).isRequired;

AboutWhatHappenedForm.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
  }).isRequired,
  data: PropTypes.shape({
    formOptions: PropTypes.shape({
      harassment_type_options: valueLabelShape,
      followup_actions_options: valueLabelShape,
      yes_no_optional_options: valueLabelShape,
    }).isRequired,
  }).isRequired,
};

export default formWrapper(loading(AboutWhatHappenedForm));