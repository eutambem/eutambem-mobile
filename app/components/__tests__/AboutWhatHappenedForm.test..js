import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import { AboutWhatHappenedForm } from '../AboutWhatHappenedForm';

const props = {
  navigation: {
    navigate: sinon.spy(),
  },
};

describe('AboutWhatHappenedForm', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<AboutWhatHappenedForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
