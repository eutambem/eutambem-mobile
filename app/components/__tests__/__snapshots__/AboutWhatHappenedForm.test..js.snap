// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutWhatHappenedForm renders correctly 1`] = `
<View>
  <PlaceInput
    navigation={
      Object {
        "navigate": [Function],
      }
    }
    onValueChange={[Function]}
    placeScreenTitle="Informe a Empresa"
    placeholder="Empresa em que ocorreu"
    required={true}
    types={
      Array [
        "establishment",
      ]
    }
  />
  <Picker
    items={
      Array [
        Object {
          "label": "Assédio Físico",
          "value": "physical",
        },
        Object {
          "label": "Assédio Verbal",
          "value": "verbal",
        },
        Object {
          "label": "Homofobia",
          "value": "homophobia",
        },
        Object {
          "label": "Racismo",
          "value": "racism",
        },
        Object {
          "label": "Transfobia",
          "value": "transfobia",
        },
        Object {
          "label": "Outro tipo de Assédio",
          "value": "other",
        },
      ]
    }
    placeholder="Tipo de assédio"
    required={true}
  />
  <TextInput
    multiline={true}
    onChangeText={[Function]}
    placeholder="Conte-nos mais sobre o ocorrido"
    required={true}
    value=""
  />
  <DatePicker
    date={null}
    maxDate={1970-01-18T14:58:09.340Z}
    onDateChange={[Function]}
    placeholder="Quando ocorreu"
    required={true}
  />
  <CheckBoxGroup
    label="Você tomou alguma providência com relação ao ocorrido?"
    options={
      Array [
        Object {
          "key": "talkedToAccountable",
          "label": "Falei com algum Responsável (RH, Superior direto, etc)",
        },
        Object {
          "key": "gaveFeedback",
          "label": "Dei o feedback para o envolvido",
        },
        Object {
          "key": "policeReport",
          "label": "Fiz um Boletim de Ocorrência",
        },
      ]
    }
  />
  <Picker
    items={
      Array [
        Object {
          "label": "Sim",
          "value": "yes",
        },
        Object {
          "label": "Não",
          "value": "no",
        },
        Object {
          "label": "Prefiro não responder",
          "value": "notInformed",
        },
      ]
    }
    placeholder="Você recomendaria essa empresa depois do ocorrido?"
    required={true}
  />
  <TextInput
    autoGrow={true}
    maxLength={350}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Qual seu conselho para os gestores?"
    required={false}
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "borderBottomColor": "#91508c",
          "borderBottomWidth": 1,
          "borderWidth": 0,
          "height": 30,
          "margin": 0,
          "marginBottom": 10,
          "marginTop": 10,
        },
        Object {
          "height": "auto",
          "minHeight": 30,
        },
      ]
    }
    value=""
  />
  <Button
    onPress={[Function]}
    title="Prosseguir"
    type="confirm"
  />
</View>
`;
